src/main/java/org/kohsuke/github/GHRelease.java

```diff
@@ -122,18 +126,36 @@ GHRelease wrap(GHRepository owner) {
      * http://stackoverflow.com/questions/12361090/server-name-indication-sni-on-java but involve more complicated
      * handling of the HTTP requests to github's API.
      *
+     * @param file file to upload
+     * @param contentType content type
+     * @param uploadUrlPrefix URL prefix for uploading; for example, for github.com it's https://uploads.github.com
+     * @return instance of GHAsset uploaded
+     * @since 1.75
      * @throws IOException
      */
-    public GHAsset uploadAsset(File file, String contentType) throws IOException {
+    public GHAsset uploadAsset(File file, String contentType, String uploadUrlPrefix) throws IOException {
         Requester builder = new Requester(owner.root);
 
-        String url = format("https://uploads.github.com%s/releases/%d/assets?name=%s",
-                owner.getApiTailUrl(""), getId(), file.getName());
+        String url = format("%s%s/releases/%d/assets?name=%s",
+                            uploadUrlPrefix, owner.getApiTailUrl(""), getId(), file.getName());
         return builder.contentType(contentType)
                 .with(new FileInputStream(file))
                 .to(url, GHAsset.class).wrap(this);
     }
 
+    /**
+     * upload the release to defaul upload URL on github.com
+     *
+     * @param file file to upload
+     * @param contentType content type
+     * @return instance of GHAsset uploaded
+     * @throws IOException
+     */
+    public GHAsset uploadAsset(File file, String contentType) throws IOException {
```

Comment:
@lanwen hi, you mean something like that?...

```java
    /**
     * Because github relies on SNI (http://en.wikipedia.org/wiki/Server_Name_Indication) this method will only work on
     * Java 7 or greater.  Options for fixing this for earlier JVMs can be found here
     * http://stackoverflow.com/questions/12361090/server-name-indication-sni-on-java but involve more complicated
     * handling of the HTTP requests to github's API.
     *
     * @param file file to upload
     * @param contentType content type
     * @param uploadUrlPrefix URL prefix for uploading; for example, for github.com it's https://uploads.github.com
     * @return instance of GHAsset uploaded
     * @since 1.75
     * @throws IOException
     */
    public GHAsset uploadAsset(File file, String contentType, String uploadUrlPrefix) throws IOException {
        Requester builder = new Requester(owner.root);

        String uploadUrlPrefix_ = uploadUrlPrefix != null ? uploadUrlPrefix : DEFAULT_UPLOAD_URL;

        String url = format("%s%s/releases/%d/assets?name=%s",
                            uploadUrlPrefix_, owner.getApiTailUrl(""), getId(), file.getName());
        return builder.contentType(contentType)
                .with(new FileInputStream(file))
                .to(url, GHAsset.class).wrap(this);
    }

    /**
     * upload the release to defaul upload URL on github.com
     *
     * @param file file to upload
     * @param contentType content type
     * @return instance of GHAsset uploaded
     * @throws IOException
     */
    public GHAsset uploadAsset(File file, String contentType) throws IOException {
        return this.uploadAsset(file, contentType, null);
    }
```